src/cmds/pbs_rsub.c

```diff
@@ -696,8 +767,131 @@ main(int argc, char *argv[], char *envp[])
 		}
 	}
 
+	if (is_maintenance_resv) {
+		int i;
+		char tmp_str[BUF_SIZE];
```

Comment:
I think 512 is acceptable. The worst case is 255 (for PBS_MAXHOSTNAME) + 10 (for max integer) + 13 (for "+host=:ncpus="), which is 278. Let me know if you still consider it to be wasting.

Reply:
hmmm. thank you for the explanation. let us keep it 512 bytes.

Reply:
The pbs_asprintf() function was introduced when we added "-Wall -Werror" to the compilation parameters. That might be another option in this case. Several buffer overrun issues were addressed with this.

Reply:
pbs_asprintf() always allocates the memory.  Here, we repeatedly snprintf to the tmp_str in the for loop. Using pbs_asprintf() instead of snprintf, we would need to free the tmp_str after each use and it would be allocated with pbs_asprintf() again. Now, calculations are spared opposed to using pbs_asprintf(). I think that the string buffer that is large enough is more suitable in this case.

Reply:
Sounds reasonable. Thank you.